db/document.go

```diff
@@ -1238,6 +1244,59 @@ func (doc *Document) MarshalWithXattrs() (data []byte, syncXattr, vvXattr, mouXa
 	return data, syncXattr, vvXattr, mouXattr, nil
 }
 
+// ConstructXattrFromHlv will build a persisted hlv from the in memory hlv. Converting the pv and mv maps to deltas
+func ConstructXattrFromHlv(hlv *HybridLogicalVector) ([]byte, error) {
+	var persistedHLV PersistedHLV
+
+	persistedHLV.SourceID = hlv.SourceID
+	persistedHLV.Version = hlv.Version
+	persistedHLV.ImportCAS = hlv.ImportCAS
+	persistedHLV.CurrentVersionCAS = hlv.CurrentVersionCAS
+	if len(hlv.PreviousVersions) > 0 {
+		persistedHLV.PreviousVersions = make([]string, len(hlv.PreviousVersions))
+		persistedHLV.PreviousVersions = VersionsToDeltas(hlv.PreviousVersions)
+	}
+	if len(hlv.MergeVersions) > 0 {
+		persistedHLV.MergeVersions = make([]string, len(hlv.MergeVersions))
+		persistedHLV.MergeVersions = VersionsToDeltas(hlv.MergeVersions)
+	}
+
+	hlvData, err := base.JSONMarshal(&persistedHLV)
+	if err != nil {
+		return nil, err
+	}
+	return hlvData, nil
+}
+
+// ParseHLVFields will parse raw bytes of the hlv xattr from persisted version to in memory version, converting the
+// deltas in pv and mv back into in memory format
+func ParseHLVFields(xattr []byte) (*HybridLogicalVector, error) {
```

Comment:
If an HLV ends up corrupted deltas, does the document just become unreadable (i.e. when this function returns error).  I feel like we need a way to repair/rewrite the HLV in this scenario (possibly omitting malformed entries in PV).

Reply:
Good shout I have not returned error in the case and continued to build HLV without the entries in this case. I chose to log at warn as if this happens its probably something the customer/we should know about? Happy to change that log level tif you have an opinion though. Added test for this pathway too `TestParseVersionVectorCorruptDelta`